### PR TITLE
docs: Indicate secret-store support

### PR DESCRIPTION
**Pull request by srebhan**
- [x] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [ ] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

This PR adds a small section about secret-store support for each related plugin. Additionally, the PR removes secret-store support for the `credentials` option in `outputs.nats` as this references a file and should not be secret.

### CODE REVIEW

**Review by powersj — COMMENTED** [review]
Did you add these via the config includes? If so is there a file that needs to be `git add`?

**Comment by srebhan** [comment]
No I added this by hand as I spell-out which options support secret-store variables. Have been the other route but this would require templating support and I wanted to avoid that...

**Review by powersj — APPROVED** [review]

File: plugins/inputs/sqlserver/README.md [hunk 1]
```diff
@@ -118,6 +118,14 @@ See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
 
 [CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
 
+## Secret-store support
+
+This plugin supports secrets from secret-stores for the `servers` option.
+See the [secret-store documentation][SECRETSTORE] for more details on how
+to use them.
+
```
Comment (Hipska) [inline plugins/inputs/sqlserver/README.md:126]: I missed this PR, but what about adding an example to some of the more complex ones?
```suggestion
This plugin supports secrets from secret-stores for the `servers` option.
 See the [secret-store documentation][SECRETSTORE] for more details on how
 to use them.

Example: `Server=192.168.1.10;Port=1433;User Id=@{my_secret_store:sqlserver_user};Password=@{my_secret_store:sqlserver_pass};app name=telegraf;log=1;`
```

Comment (srebhan) [reply plugins/inputs/sqlserver/README.md:126]: Feel free to put up a PR!

**Pull request merged**